### DISCUSSION TITLE
Help for components polarity

### DISCUSSION THREAD

**Discussion by Bebowl** [will127534/OneInchEye#19] [Q&A]
Hi Will, may I ask you help to confirm polarity of U4, U5, U7 and U8 on attached picture.
Thanks 
Ben
![84648054886400-Produce_DanZhi SMT_Snapshot Top 3265289A_Y9 SMT02407041822701](https://github.com/will127534/OneInchEye/assets/174592155/db5f316d-5c87-4e38-a31e-d58bd0a02ef8)

**Comment by will127534** [comment]
Yeap, looks good

**Comment by will127534** [comment]
Before I close this discussion, just a remember to jump the EN pin (JP2) and pull-down on the EN pin (JP1 - L) when you get the board back.

**Comment by Bebowl** [comment]
Oh yes I see, like you did on the tindle picture.
Thanks for the warning. I will keep you in touch when I get the board. Really excited to receive it.
![2024-01-07T08_54_28 828Z-_DSC4023](https://github.com/will127534/OneInchEye/assets/174592155/38638f40-71aa-440f-ad83-4439ab7d8139)

**Reply by Bebowl** [reply]
I have tried to find what are this jumpers for, but did not find any information. Can you give some details?
Thanks

**Comment by will127534** [comment]
Have you check the most important documents - the schematic? 
This is for connecting the enable pin on the FPC, in some cases (specifically my all-sky camera project) may need to ignore the Enable signal. That's why there is also a pull-up and down selector jumper on the board in case enable signal is not used.

**Comment by Bebowl** [comment]
Hi Will, I am still waiting the board, but have two more questions. Which json tuning file is used for the IMX283. And a second question, after following the installation steps of the Quick Start Guide, I cannot see any imx283.dtbo file under overlays directory. I was wondering if it is normal ?
Thanks for your help

Ben

**Reply by will127534** [reply]
quick question: you do see a imx283.dtbo in the local directory right?

**Comment by Bebowl** [comment]
I found a imx283.dtbo under /var/lib/dkms/imx283/0.0.1/source, so it seems to work. I think I will have to move it to overlays directory ?
I receive some errors while running the setup.sh script : 
Error! Your kernel headers for kernel 6.6.31+rpt-rpi-v8 cannot be found at /lib/modules/6.6.31+rpt-rpi-v8/build or /lib/modules/6.6.31+rpt-rpi-v8/source

And I also found the imx283.json file.

**Comment by Bebowl** [comment]
Ok I have made a new fresh reinstall of OS and now installation works fine. Now waiting for the camera board to arrive.

**Comment by Bebowl** [comment]
Just received two boards from JLCPCB, I have jumped JP1-L and JP2, connect to the Rpi5 with FPC cable, but... unfortunately the Rpi5 does not start, the led stays red, meaning hardware issue. I have checked, as far as I can, the hardware and electronic seem ok, but hard to find what's wrong... When unplug, the Rpi5 starts correctly.

**Reply by will127534** [reply]
you will have to find where is the shortcut happening, try testing the 1.2V/2.9V/1.8V point to see which is causing the short.
Also double check the IMX283 orientation

**Comment by Bebowl** [comment]
Thanks Will, I will test that. I have also one board where JP1-L and JP2 are not soldered. If I plug it to Rpi will it also generate hardware issue (to be sure I don't make a bad solder on jp1 and jp2) ?

**Reply by will127534** [reply]
EN pin are not connected -> the voltage rail won't be on

**Comment by Bebowl** [comment]
That's what was ordered to JLCPCB, seems correct... I will make test on voltage plugs as you suggested and if not conclusive will unsolder the IMX283 to check if polarity is correct. I will keep you in touch. 
Still many thanks for your help. I would be so glad that it could work !! If you have one in stock for sell it would also be nice to make test on my project.

![image](https://github.com/user-attachments/assets/d76c962a-43ec-499d-b9ff-9336050fefd4)

![image](https://github.com/user-attachments/assets/661af17d-bcf4-4ecc-a3b7-cb3f72828fc2)

**Reply by Bebowl** [reply]
Also xray report from JLCPCB and it seems that IMX283 is correctly oriented

![1](https://github.com/user-attachments/assets/c0e30bd0-bcca-4b71-965b-37a1eebb7d81)

**Comment by Bebowl** [comment]
Raspberry does not start, so voltage is not delivered to camera board and I am not able to test 1.2V/2.9V/1.8V... I am about to unsolder the IMX283 and see if orientation is correct, but it seems also to be ok... Again, if you have one board for sell I would be glad to buy you one, test my project and maybe understand what is not working on mine.

**Comment by will127534** [comment]
Have you check the FPC yet?

**Reply by Bebowl** [reply]
Hi Will, yes when I start the Rpi5 without the sensor board, I keep the FPC plug and Rpi5 starts correctly. I have also tested the gnd and 3.3v on the FPC and it is ok. Also try another FPC and it is the same.

**Reply by will127534** [reply]
I mean, when you connect the FPC to camera sensor board, did you connect it in the way 3.3V matches with 3.3V pin on the FPC?

**Reply by will127534** [reply]
Also using a DMM can identify if you have a short on vcc rails for the sensor or short on the input 3.3V better. If you connect the board without EN pin connected, voltage regulators won't even enable. So I'm guessing you are not connecting the FPC in the right position or the FPC connector you used is not the same type as the bom.

**Reply by Bebowl** [reply]
The Rpi5 does not start when board is connected, it doesn't deliver anymore voltage to the board and DMM cannot be used.

**Comment by Bebowl** [comment]
Hi Will, I really order the same FPC as the one in your bom
![Screenshot_2024-07-21-10-16-55-450_com android chrome-edit](https://github.com/user-attachments/assets/291461d3-54bd-4ee7-af6e-16a486263168)

**Comment by Bebowl** [comment]
Attached is a picture of my board with pins soldered. The FPC cable is inserted as indicated in quick guide. 
![IMG_20240721_101137](https://github.com/user-attachments/assets/fe633592-7d43-49bc-9f56-70561dfa2569)

**Reply by will127534** [reply]
I see, yeap that looks good to me.

Can you check the 3.3V is shorted or not?

**Reply by Bebowl** [reply]
When I test the FPC cable with DMM I see that 3.3v works, but as the Rpi5 does not start when board is connected the DMM cannot be used.

Btw thanks for all your time and assistance, helping on this subject through the net is quite complicated. But I must say that being able to find what is wrong would be a big big step for my project.

**Reply by will127534** [reply]
I don't understand why you say DMM can not be used, can't you check if the board has shorted with the DMM with or without powering up the board?

**Reply by Bebowl** [reply]
Hi Will, my mistake. I have tested continuity on several points, not yet found the issue. But don't understand how I am supposed to test the 1.2V/2.9V/3.3V points ? Do I put one DMM end point to this test point and the other to a GND point ?

**Reply by will127534** [reply]
Yes

**Reply by will127534** [reply]
On the same time, can you take a photo of your RPI5 and show me how you connect the FPC between the two?

**Comment by Bebowl** [comment]
Hi Will attached is a picture of FPC on Rpi5 side.
I have also tested 1.2V/2.9V/3.3V points with GND point below 3.3v point with FPC cable removed and get following results on DMM (red DMM cable on GND and black cable on test points) : 188, 275, 458.
![IMG_20240722_075720](https://github.com/user-attachments/assets/33f5d503-8488-4ec3-81c9-b678295f7752)

**Reply by Bebowl** [reply]
I just get a better DMM with continuity alarm (previous only has diode test), and I confirm no continuity on voltage test points and gnd test point.

**Reply by will127534** [reply]
Then it shouldn't really cause the RPI5 not booting up... when you connect it to RPI and try to boot up by button press, do you see 3.3V showing up on the sensor board for a sec?

**Reply by Bebowl** [reply]
Nope cannot see any 3.3V showing on the sensor board. 
I have tested with DMM the FPC cable connected to the RPI but not connected to sensor board and it correctly delivered 3.3v, even when sd card is removed and no OS is started. 
I have also tested a sensor board where neither JP1 nor JP2 is soledered and same result RPI does not start and led stays red on RPI...

**Reply by will127534** [reply]
no continuity on voltage test points - including 3.3V ?

**Reply by Bebowl** [reply]
No continuity also on 3.3v test point. I wonder if the fpc plug is not the problem... But sounds quite improbable that soldering from JLCPCB was not good

**Reply by will127534** [reply]
Honestly I don't know what happened because if the 3.3V FPC is not shorted on the board, you should not have 3.3V shortcut when you connect it to RPI.

**Reply by Bebowl** [reply]
Thanks Will for your help. I will continue to investigate, as digging help me also to understand architecture of your board (understand now that some early questions were stupid).
I can ship you this board if you want and if you have one available to buy I am also interested.

**Comment by Bebowl** [comment]
Will my dear friend !!!! At least it works !!! 

I have bought a new set of fpc 22 pin flat cables, really not thinking it was the issue, because I have tested several fpc 22 pin cables. However all my fpc cables were coming from Arducam. So I don't know exactly what they did with their cables but they are not compatible with your cam.

Again and again many thanks for your patience and all replies. I was suspecting a very stupid issue but not so stupid.

So ready for test now !!

**Reply by will127534** [reply]
Congratulations! 
Can you take a photo of the FPC cable you were using? I never see this failure mode.

**Reply by Bebowl** [reply]
Attached is a picture of the cable. On the left the one which does not work and on the right the one that works. They are reversed.

By the way have you tested the Picamera python library with your camera ?

![IMG_20240726_074904](https://github.com/user-attachments/assets/f04a09f4-5c34-454b-a0af-5663b376c395)

**Reply by will127534** [reply]
I see, I have the notes on the StarlightEye for the users to use single sided FPC cables but I should add that back to OneIncheye.
Yes, I've used picamera2 for my all sky camera, but I think you will need to move some files so the python library uses the correct libcamera.o. I got the same question from FourthirdEye project so will probably setup a fresh install to test that out and see what are the steps this weekend.

**Reply by Bebowl** [reply]
Yes I see that picamera2 is pointing to /usr/lib/aarch64-linux-gnu/libcamera rather than /usr/local/lib/aarch64-linux-gnu/libcamera.

I can move files but wonder if there is not a smarter way to do that. I ask the question on rpi forum.

**Reply by will127534** [reply]
I typically just move the files manually

**Comment by Bebowl** [comment]
Hi Will, hope you're well. Testings of Oneincheye work great. Thanks again for all your work and help.
Just a small question, when I get the image from sensor it seems that I have several lines of pixels vertical and horizontal at the edge of the image that do not display image (like if they were dead). I solve the problem by cropping the image, but was wondering if it is normal (are they some kind of services pixels) ?
